### DISCUSSION TITLE
3-body term using pkron

### DISCUSSION THREAD

**Discussion by ProThicc** [jcmgray/quimb#213] [General]
Hello,
I am trying to define a 3-body hamiltonian, but instead of using ikron as in ham_heis, I used pkron since placing 3-body operators was much more intuitive in pkron at different indices.
However when using the ownership argument with mpi, I noticed it wasn't doing the same thing as ikron. I think that is not implemented yet. (shows up in TODO)
Is there a workaround to place operators at arbitrary indices using ikron itself which I can use until pkron is implemented? 

Thanks

**Comment by jcmgray** [comment]
Hi @ProThicc, the current best answer I think would be to break the 3-body operators into a sum of 3 single site tensor products (e.g. for heisenberg model $S \cdot S = XX + YY + ZZ$), then use `ikron` as usual and sum over terms at the top level. See `ham_heis_2D` for an example. If the decomposition is not symbolically obvious then you might have to use SVD to due the decomposition spatially.

Long-term there is the `quimb.experimental.operatorbuilder` module which will probably replace ikron and should be more efficient and support ownership more naturally etc.

**Reply by ProThicc** [reply]
Hello, @jcmgray, Thank you so much! I implemented this with ikron looking at the ham_heis_2D example and it worked well!
